chain/jsonrpc/tests/test_utils.rs

```diff
@@ -56,3 +56,22 @@ pub fn start_all_with_validity_period(
     );
     (view_client_addr, addr)
 }
+
+#[allow(unused_macros)] // Suppress Rustc warnings even though this macro is used.
+macro_rules! test_with_client {
+    ($validator_status:expr, $client:ident, $block:expr) => {
+        init_test_logger();
+
+        System::run(|| {
+            let (_view_client_addr, addr) = test_utils::start_all($validator_status == "validator");
```

Comment:
this feels a bit weird. Why not just use a boolean?

Reply:
If I use boolean than the usage will be:

```rust
test_with_client!(false, client, ...);
```

This `false` is not self-descriptive and I have to look it up. I may introduce an enum, though.

Reply:
Right but I feel like your way is no better than using boolean. It was a bit confusing to me at first. I think definitely enum would be better if you want to be more explicit.